vtex-apps/admin-pages
# Font Family Editor

#### What is the purpose of this pull request?
To include a font family editor on the styles editor.
This allows admins to save and delete their customised font families.

This PR does not include the UI for them to choose which fonts will style the website.

#### How should this be manually tested?
The changes on this PR can be accessed on:
https://athos--storecomponents.myvtexdev.com/admin/cms/storefront

To test it, edit any style, go to "Typography", then "Font Family".
There you can add a new font family or edit an existing one.

When editing (or adding) a new font family, you need to add a not empty font family name,
and at least one font family file (with valid file extension and <= 2MB). Each file must have a "style" selected (which is actually a style + weight pair) before saving the changes.

If you try to upload an invalid file, a toast should let you know.
Toasts also tell you when a save and a delete are successful.

Loading states are signalised by spinners and currently there is no error state.


#### Screenshots or example usage
Here is the flow a user should go through to add new fonts:

<img width="872" alt="Screen Shot 2019-05-09 at 00 08 23" src="https://user-images.githubusercontent.com/7853668/57424792-4b8a7b80-71ef-11e9-9027-8a76f66120f6.png">
<img width="948" alt="Screen Shot 2019-05-09 at 00 08 28" src="https://user-images.githubusercontent.com/7853668/57424793-4c231200-71ef-11e9-8e14-5051d25a08fa.png">
<img width="1019" alt="Screen Shot 2019-05-09 at 00 08 33" src="https://user-images.githubusercontent.com/7853668/57424794-4c231200-71ef-11e9-90bc-7e40f43bc335.png">
<img width="958" alt="Screen Shot 2019-05-09 at 00 08 38" src="https://user-images.githubusercontent.com/7853668/57424795-4c231200-71ef-11e9-86f2-a16494525f8b.png">

<img width="964" alt="Screen Shot 2019-05-09 at 00 08 57" src="https://user-images.githubusercontent.com/7853668/57424791-4b8a7b80-71ef-11e9-83bc-6112a5a32ce7.png">
<img width="955" alt="Screen Shot 2019-05-09 at 00 09 06" src="https://user-images.githubusercontent.com/7853668/57424790-4b8a7b80-71ef-11e9-8bf5-e0fb010334bb.png">
<img width="903" alt="Screen Shot 2019-05-09 at 00 09 12" src="https://user-images.githubusercontent.com/7853668/57424788-4b8a7b80-71ef-11e9-8996-219f28f93c63.png">


#### Types of changes
